segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -454,7 +455,21 @@ public void commitTransaction(CommitTransaction commitTx) {
                 .thenAccept(v -> connection.send(new TransactionCommitted(requestId, commitTx.getSegment(), commitTx.getTxid())));
 
         CompletableFuture.allOf(seal, merge)
-                .exceptionally(e -> handleException(requestId, transactionName, "Commit transaction", e));
+                .exceptionally(e -> {
+                    final Throwable cause;
+                    if (e instanceof CompletionException) {
+                        cause = e.getCause();
+                    } else {
+                        cause = e;
+                    }
+                    if (cause instanceof StreamSegmentMergedException) {
```

Comment:
I think we said offline that it is better not to create a new response type and handle `StreamSegmentMerged` as a confirmation that the commit has already happened, but now looking at the flow of other operations, we should really be simply calling `handleException`in exceptionally` for symmetry and as such we would need a new return type. What do you think?

Reply:
Ok, I can see that we handle `abortTransaction` in the following way:

```
.exceptionally(e -> {
                    if (Exceptions.unwrap(e) instanceof StreamSegmentNotExistsException) {
                        connection.send(new TransactionAborted(requestId, abortTx.getSegment(), abortTx.getTxid()));
                        return null;
                    } else {
                        return handleException(requestId, transactionName, "Abort transaction", e);
                    }
                });
```

what we are doing here isn't very different. can we handle how we unwrap the exception the same way for symmetry?

Reply:
sure :)